server/public/model/post_metadata.go

```diff
@@ -30,6 +30,23 @@ type PostMetadata struct {
 	Acknowledgements []*PostAcknowledgement `json:"acknowledgements,omitempty"`
 }
 
+func (pm *PostMetadata) Auditable() map[string]interface{} {
+	var embeds []map[string]any
+	for _, pe := range pm.Embeds {
+		embeds = append(embeds, pe.Auditable())
+	}
```

Comment:
The problem with that is the field is marked `omitempty` and instead of letting a nil stay nil, it will use an empty map.  I guess I can add a conditional that nils it again.

Reply:
I see. Either that or we can declare the variable first and then allocate and do the loop under an if condition.